board/src/main/java/com/main/board/post/service/PostService.java

```diff
@@ -0,0 +1,212 @@
+
+        return new PostDetailResponse(postDetailList, commentList, imageUrlList);
+    }
+
+    public List<MoreCommentResponse> getRecursiveMoreComment(long commentId, long offset, long postId) {
+        //1. 반환 객체 생성
+        List<MoreCommentResponse> moreCommentResponseList = new ArrayList<>();
+        //2. 대댓글 정보 가져오기
+        List<CommentDetailFromDB> commentList = postRepository.getRecursiveMoreCommentList(commentId, offset, postId);
+
+        Map<Long, CommentDetailFromDB> commentMap = new HashMap<>();
+
+
+        // 3. 대댓글 데이터를 MoreCommentResponse로 변환하여 리스트에 추가
+        for (CommentDetailFromDB comment : commentList) {
+            commentMap.put(comment.getCommentId(), comment);
+        }
+
+        for(CommentDetailFromDB comment : commentList) {
+            if(comment.getParentId() != null) {
+                CommentDetailFromDB parent = commentMap.get(comment.getParentId());
+                if(parent != null) {
+                    parent.getChildCommentList().add(comment);
+                }
+                else {
+                    MoreCommentResponse convertingData = new MoreCommentConverter(comment).toObject();
+                    moreCommentResponseList.add(convertingData);
+                }
+            } else {
+                MoreCommentResponse convertingData = new MoreCommentConverter(comment).toObject();
+                moreCommentResponseList.add(convertingData);
+            }
+        }
+
+        return moreCommentResponseList;
+    }
+    //recursive 방식 끝
+
+
+    @Transactional
+    public void createPost(CreatePostRequest createPostRequest) {
+        if(createPostRequest.getPostImageList() == null) {
+            postRepository.createPost(createPostRequest);
+        }
+        else {
+            //파일 업로드 로직
+            String filePath;
+            try {
+                //1. 게시물 등록
+                postRepository.createPost(createPostRequest);
+                //2. 파일 경로 DB에 저장하기위해 게시물 조회
+                long recentlyPostId = postRepository.selectRecentPostId(createPostRequest.getMemberId());
+
+                for(int i = 0; i < createPostRequest.getPostImageList().size(); i++) {
+                    String fileName = UUID.randomUUID() + "_" + createPostRequest.getPostImageList().get(i).getOriginalFilename();
+                    filePath = uploadDir + "\\" + fileName;
+
+                    //1. 파일 업로드
+                    createPostRequest.getPostImageList().get(i).transferTo(new java.io.File(filePath));
+                    //2. DB에 파일 경로 저장
+                    postRepository.createPostImage(recentlyPostId, filePath);
+                }
+
+            } catch (Exception e) {
+                throw new RuntimeException("파일 업로드 실패", e);
+            }
+        }
+    }
+
+    public void updatePost(UpdatePostRequest updatePostRequest) {
+        boolean checkOwner = postRepository.selectPostForMember(updatePostRequest.getPostId(), updatePostRequest.getMemberId());
```

Comment:
post정보를 가져와서 check해주는 방식이 좀 더 간결하지 않을까요?
```suggestion
        boolean checkOwner = loginUser.getMemberId().equals(postRepository.findById(updatePostRequest.getPostId()).getMemberId());
```

Reply:
아 이부분 멘터링때 질문했어야했는데 깜박했습니다...
이거 어떤의도이신건지 잘 모르겠어서 이해가 잘안갑니다...

Reply:
`postRepository.selectPostForMember(updatePostRequest.getPostId(), updatePostRequest.getMemberId());`는 쿼리에서 바로 boolean값이 떨어지는 형태라서 이 boolean값이 어떤 값인지 유추하기가 어렵고, 실제로 쿼리까지 들어가서 어떤 쿼리인지 보아야 boolean값의 정체를 알 수 있는 형태입니다.

이를 아예 직관적으로 `loginUser.getMemberId().equals(postRepository.findById(updatePostRequest.getPostId()).getMemberId()` 형태로 작성해주면 boolean값이 어떤 값인지 명시적으로 알 수 있습니다.